lighthouse-core/audits/large-javascript-libraries.js

```diff
@@ -0,0 +1,174 @@
+  description: 'Large JavaScript libraries can lead to poor performance. ' +
+    'Prefer smaller, functionally equivalent libraries to reduce your bundle size.' +
+    ' [Learn more](https://developers.google.com/web/fundamentals/performance/webpack/decrease-frontend-size#optimize_dependencies).',
+  /** Label for a column in a data table. Entries will be names of large JavaScript libraries that could be replaced. */
+  name: 'Library',
+  /** [ICU Syntax] Label for the Large JavaScrip Libraries audit identifying how many large libraries were found. */
+  displayValue: `{libraryCount, plural,
+    =1 {1 large library found}
+    other {# large libraries found}
+    }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LargeJavascriptLibraries extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'large-javascript-libraries',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['Stacks'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {Array<{original: MinifiedBundlePhobiaLibrary, suggestions: MinifiedBundlePhobiaLibrary[]}>} */
+    const libraryPairings = [];
+    const detectedLibs = artifacts.Stacks.filter(stack => stack.detector === 'js');
+
+    const seenLibraries = new Set();
+
+    for (const detectedLib of detectedLibs) {
+      if (!detectedLib.npm || !libStats[detectedLib.npm]) continue;
+      const suggestions = librarySuggestions[detectedLib.npm] || [];
+
+      if (seenLibraries.has(detectedLib.npm)) continue;
+      seenLibraries.add(detectedLib.npm);
+
+      let version = 'latest';
+      if (detectedLib.version && libStats[detectedLib.npm].versions[detectedLib.version]) {
+        version = detectedLib.version;
+      }
+
+      const originalLib = libStats[detectedLib.npm].versions[version];
+
+      /** @type {Array<{name: string, repository: string, gzip: number}>} */
+      // @ts-ignore
+      let smallerSuggestions = suggestions.map(suggestion => {
+        // @ts-ignore
+        if (libStats[suggestion]['latest'].gzip > originalLib.gzip) return;
+
+        return {
+          name: suggestion,
+          repository: libStats[suggestion].repository,
+          // @ts-ignore
+          gzip: libStats[suggestion]['latest'].gzip,
+        };
+      });
+
+      smallerSuggestions = smallerSuggestions.sort((a, b) => a.gzip - b.gzip);
+      if (smallerSuggestions.length) {
+        libraryPairings.push({
+          original: {
+            // @ts-ignore
+            gzip: originalLib.gzip,
+            name: detectedLib.npm,
+            // @ts-ignore
+            repository: libStats[detectedLib.npm].repository,
+          },
+          suggestions: smallerSuggestions,
+        });
+      }
+    }
+
+    const tableDetails = [];
```

Comment:
might be worth either `.map` or adding a type here to avoid devolving into `any[]`

![image](https://user-images.githubusercontent.com/2301202/89202533-b48aab80-d578-11ea-9df2-832e2d7a9ba6.png)

![image](https://user-images.githubusercontent.com/2301202/89202565-c4a28b00-d578-11ea-9e8a-fde8f9fe47f9.png)